### ISSUE TITLE
Proxy minion never comes up with message ' I am XXX and I am not supposed to start any proxies.'

### ISSUE THREAD

**Issue by hishailesh77** [saltstack/salt#25908]
I have been trying to explore the proxy minion feature with REST and perform all the configuration for REST as per the instructions give at 'http://docs.saltstack.com/en/latest/topics/topology/proxyminion/index.html' .
But proxy minion never comes up when i start the actual minion, and i always get the message 
"I am XXX and I am not supposed to start any proxies"

And here is my sample pillar configuration for REST_SAMPLE
```yaml
root @ docker: / srv / pillar # cat top.sls
base:
  docker:
    - Proxyminion
```
```yaml
root @ docker: / srv / pillar # cat proxyminion.sls
proxy:
  rest_sample:
    proxytype: rest_sample    
    url: http://127.0.0.1:8080/
    id: proxy_docker
```

**Comment by jfindlay** [comment]
@hishailesh77, thanks for the report.

**Comment by cro** [comment]
@hishailesh77 Good morning.  Fixes for this, including more accurate documentation for the rest_sample example will be committed to the 2015.5 branch later today.  2015.8 and develop will take a little longer because the TCP transport changes are going to require some refactoring of the proxy.

**Comment by hishailesh77** [comment]
Thanks CR .

**Comment by rallytime** [comment]
2015.5 branch changes/fixes are in #26135

**Comment by hishailesh77** [comment]
I have installed salt (master and minion) from 2015.5 branch , but i am still getting the message " I am XXX and I am not supposed to start any proxies. (Likely not a problem)".
In order to install salt, I  have followed the instructions on 'http://docs.saltstack.com/en/latest/topics/development/hacking.html' for '2015.5' Git Branch  .


.
